lib/simulator/simulatormodule.c

```diff
@@ -952,7 +952,25 @@ static PyObject *deregister_callback(PyObject *self, PyObject *args)
     gstate = TAKE_GIL();
 
     pSihHdl = PyTuple_GetItem(args, 0);
-    hdl = (gpi_sim_hdl)PyLong_AsUnsignedLong(pSihHdl);
```

Comment:
I'm not sure that this is needed. We already work with 32 bit simulators using the existing code and a 32 bit python. In which case long is 4 bytes already, and 8 on 64 bit.